### ISSUE TITLE
@Deprecated to work like @Ignore

### ISSUE THREAD

**Issue by phillip-kruger** [eclipse/microprofile-graphql#47]
Add test cases for @Deprecated and make sure that it can work the same as @Ignore

Meaning that if placed on the field, both Input and Output is deprecated, however putting it on the getter of setter only the input or output is deprecated. (Same as @Ignore)

**Comment by andymc12** [comment]
This was considered a stretch goal for 1.0, but we're past the original cutoff date for 1.0, so I'm moving this to 1.1.  We will need to remove the existing `@Deprecated` API annotation and spec text for 1.0 - I'll open a separate issue or that.

**Comment by phillip-kruger** [comment]
Can this now move back to 1.0 ?

**Comment by t1** [comment]
Actually, I think things annotated as `@Deprecated` should not be simply ignored, but get the [`@deprecated`](http://spec.graphql.org/draft/#sec--deprecated) directive instead.

**Comment by phillip-kruger** [comment]
Yes, that is the idea, the "work like ignore" just means that we allow it on get and set and field, so it applies to output, input or both. We will also have our own annotation (that can take a reason) but will also support the java one.

**Comment by t1** [comment]
The new annotation could be implemented as a custom directive #344.
